File: mcst/ClassDef.m
classdef ClassDef < Segment
    properties
        head Statement
        property Properties
        method Methods
        end_ Statement
    end
    methods
        function self = ClassDef(head, property, method, end_)
            self.head = head;
            self.property = property;
            self.method = method;
            self.end_ = end_;
        end
    end
end

File: test_m/py/nodes/ClassDef.py
from mruntime import *
from test_m.py.nodes.Segment import Segment
class ClassDef(Segment):
    def __init__(self, *nargin): # retval: self
        [head, property, method, end_] = nargin
        nargin = len(nargin)
        self.head = head
        self.property = property
        self.method = method
        self.end_ = end_
    @staticmethod
    def empty():
        return []
